File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anag_map = defaultdict(list)

        for word in strs:
            sorted_word = ''.join(sorted(word))
            anag_map[sorted_word].append(word)
        return list(anag_map.values())

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        if (strs == null || strs.length == 0) return new ArrayList<>();

        Map<String, List<String>> anagrams = new HashMap<>();

        for (String word: strs){

            char[] sortWord = word.toCharArray();
            Arrays.sort(sortWord);
            String sorted = new String(sortWord);

            if(!anagrams.containsKey(sorted)){
                anagrams.put(sorted, new ArrayList<>());
            }

            anagrams.get(sorted).add(word);
        }
        return new ArrayList<>(anagrams.values());
    }
}